cmake/modules/AddSwift.cmake

```diff
@@ -286,9 +286,7 @@ function(_add_host_variant_c_compile_flags target)
 
   if(SWIFT_HOST_VARIANT_SDK STREQUAL "LINUX")
     if(SWIFT_HOST_VARIANT_ARCH STREQUAL x86_64)
-      # this is the minimum architecture that supports 16 byte CAS, which is
-      # necessary to avoid a dependency to libatomic
-      target_compile_options(${target} PRIVATE -march=core2)
+      target_compile_options(${target} PRIVATE -mcx16)
```

Comment:
Can you re-add the comment here and explain that -mcx16 adds support for the 16 byte cas.

Reply:
I would also explain why we are setting this flag vs a -march flag (that we are allowing for old machines to use swift).

Reply:
Done; tried to keep the comment terse but useful.

Reply:
LGTM. Thanks!